pyccel/codegen/printing/cwrappercode.py

```diff
@@ -159,7 +198,170 @@ def get_cast_function_call(self, cast_type, arg):
 
         return FunctionCall(cast_function, [arg])
 
-    def get_PyArgParseType(self, used_names, variable, argument):
+    # -------------------------------------------------------------------
+    # Functions managing  the creation of wrapper body
+    # -------------------------------------------------------------------
+    def _body_optional_variable(self, tmp_variable, variable, collect_var, check_type = False):
+        """
+        Responsible for collecting value and managing error and create the body
+        of optional arguments in format
+                if (pyobject == Py_None){
+                    collect Null
+                }else if(Type Check == False){
+                    Print TypeError Wrong Type
+                    return Null
+                }else{
+                    assigne pyobject value to tmp variable
+                    collect the adress of the tmp variable
+                }
+        Parameters:
+        ----------
+        tmp_variable : Variable
+            The temporary variable  to hold result
+        Variable : Variable
+            The optional variable
+        collect_var : variable
+            the pyobject type variable  holder of value
+        check_type : Boolean
+            True if the type is needed
+
+        Returns
+        -------
+        body : list
+            A list of statements
+        """
+        body = [(PyccelEq(VariableAddress(collect_var), VariableAddress(Py_None)),
+                [Assign(VariableAddress(variable), variable.value)])]
```

Comment:
Does this translate to:
```
int* a;
a = 4;
```
? Or did I miss something

Reply:
Do we need to allocate memory for this pointer?

Reply:
Optional variable here value is NULL so its translated to :  
```
int *a;
a = NULL;
```

Reply:
Ah of course. I don't think this is the first time I've made this mistake. Maybe it would be better to put `Nil()` then it's clear